glommio/src/sys/uring.rs

```diff
@@ -1545,4 +1567,23 @@ mod tests {
             None => {}
         }
     }
+
+    #[test]
+    #[should_panic(expected = "Unterminated SQE link chain")]
```

Comment:
what about a test for successful chaining ? 

Another interesting case is what happens when the chain is bigger than the the amount of SQEs the ring is configured for.

If I read your code correctly we'll end up waiting forever and hanging in that case. Erroring out the entire chain would be better in that case (granted this is very, very unlikely especially right now. But once eBPF lands into io_uring, I expect user-defined chains that can get arbitrarily long)

Reply:
For successful chaining, I was having trouble coming up with a good small test case (and I hoped the `connect_timeout` tests would partially cover it). The too-large chain is a good one, I think you're right about it hanging since `prepare_sqes` should fail over and over. I'm not sure if there's an easy way to get queue sizes.

Reply:
the queue size is determined at creation. Right now I am open coding it, you can just make it into a const.

Reply:
Thank you for reviewing, I added this in ca28b7e!

Reply:
There is now a test for successful links in 66e61fa